src/hooks/index.jsx

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect } from 'react';
+import { initialize, mergeConfig, getConfig } from '@edx/frontend-platform';
+
+export const EDX_HEADER_COOKIE_PAYLOAD = 'edx-jwt-cookie-header-payload';
+
+function getCookie(name) {
+  return document.cookie
+    .split('; ')
+    .map(c => c.split('='))
+    .find(([key]) => key === name)?.[1] || null;
+}
+
+function useGetMenuOptionsByRole(appID) {
+  const [userLinks, setUserLinks] = useState([]);
+
+  useEffect(() => {
+    initialize({
+      messages: [],
+      requireAuthenticatedUser: true,
+      handlers: {
+        config: () => {
+          const configOptions = {
+            MFE_CONFIG_API_URL: `${process.env.LMS_BASE_URL}/api/mfe_config/v1`,
+          };
+
+          if (appID) {
+            configOptions.APP_ID = appID;
+          }
+
+          mergeConfig(configOptions);
+        },
+      },
+    });
+  }, [appID]);
+
+  const instructorPath = getConfig().INSTRUCTOR_PORTAL_PATH;
+  const institutionPath = getConfig().INSTITUTION_PORTAL_PATH;
+
+  useEffect(() => {
+    const CERTPREP_MANAGER = {
+      type: 'item',
+      href: institutionPath,
+      content: 'Pearson Skilling Administrator',
+    };
+
+    const INSTRUCTOR_PORTAL = {
+      type: 'item',
+      href: instructorPath,
+      content: 'Pearson Skilling Instructor',
+    };
+
+    const ROLES_PERMISSIONS = {
+      GLOBAL_STAFF: [CERTPREP_MANAGER, INSTRUCTOR_PORTAL],
+      INSTITUTION_ADMIN: [CERTPREP_MANAGER, INSTRUCTOR_PORTAL],
```

Comment:
Only if the global staff and institution admin have the 'instructor' role could see the option